process/smartContract/vmInput.go

```diff
@@ -19,7 +19,9 @@ func (sc *scProcessor) createVMDeployInput(tx data.TransactionHandler) (*vmcommo
 
 	vmCreateInput := &vmcommon.ContractCreateInput{}
 	vmCreateInput.ContractCode = deployData.Code
-	vmCreateInput.ContractCodeMetadata = deployData.CodeMetadata.ToBytes()
+	// when executing SC deploys we should always apply the flags
+	codeMetadata := sc.blockChainHook.ApplyFiltersOnCodeMetadata(deployData.CodeMetadata)
```

Comment:
`deployData.CodeMetadata` is parsed on the first line of this function. The parser that does this will _**not return an error**_ if the bytes are wrong, it will just return an empty `CodeMetadata{}`. Neither the VM nor the SCProcessor will be able to distinguish between user intention or user mistake.

We should change the function `CodeMetadataFromBytes()` in `elrond-vm-common` to also return an error if the bytes are wrong. Otherwise, the empty `CodeMetadata{}` will propagate through the VM and will be saved by the SCProcessor as if that's what the user intended.

Unfortunately, this case will not be caught by the verifications finally applied on the `CodeMetadata` contained by the resulting `VMOutput`, as we discussed.

Reply:
I know. Will postpone the change for the Deploy process for now